### PR TITLE
Add npm scripts

### LINKED ISSUE #558: Package.json and npm changes
There some things that we can change with `npm`. They are:
### 1 - Dependencies x dev-dependencies
- [ ]  The dependencies that is in package.json are not `dependencies` of the project. Its use only for development. So, no make sence. 
  selection_376
### 2 - Don't need a global install of `jscs jshint jsdoc`
- [ ] Change it to only `npm install`
  selection_378
### 3 - It can use more npm scripts
- [ ] Why `./skulpt.py dist` is not a npm script? We can centralize all in package.json scripts and run `npm build` or something like that.

### PR DESCRIPTION
**Pull request by j-f1**
Fixes #558.

* Removes the recommendation to `npm install -g`
* Adds npm scripts for the following commands:
    ```
    testdebug, regengooglocs, regensymtabtests, rununits, vmwareregr, regenparser, regenasttests, regenruntests, upload, doctest, doc, browser, debugbrowser, vfs, host, repl, test
    ```
* `npm run build` runs `./skulpt.py dist`

**Warning**: Passing parameters directly to `npm` scripts doesn’t work. Instead, use `--` and then pass the options.

### CODE REVIEW

**Comment by albertjan** [comment]
How are you running jscs and jshint now?

**Comment by j-f1** [comment]
@albertjan https://github.com/j-f1/skulpt/blob/b5239c09755f343344668a681c293fe7e77059cc/skulpt.py#L226-L237

**Comment by albertjan** [comment]
and we don't need global installs for this?

**Comment by j-f1** [comment]
@albertjan `npm run` adds `node_modules/.bin` to the `PATH`, and `jscs`/`jshint` should put their binaries in that directory automatically.

**Comment by albertjan** [comment]
Ah really cool, we learn everyday 😄 And I agree we don't have to install them globally anymore on travis. Locally we'll still run `skulpt` probably, because thats engrained in my muscle memory.

We do have to maintain the scripts in two places.

File: README.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ Building Skulpt is straightforward:
 
 1. Clone the repository from GitHub, ideally using your own fork if you're planning on making any contributions
 2. Install node.js
-3. Install the jscs, jshint and jsdoc node modules using `npm install -g jscs jshint jsdoc` (you may need to use `sudo` to run this command)
+3. Install the required dependencies using `npm install`
 4. Navigate to the repository and run `./skulpt.py dist`
```
Comment (albertjan) [inline README.md]: shouldn't this also say `npm run build`

Comment (j-f1) [inline README.md]: @albertjan huh? https://github.com/skulpt/skulpt/pull/664/files#diff-04c6e90faac2675aa89e2176d2eec7d8R57

**Comment by albertjan** [comment]
Might be a good idea to add a `npm run dist-min` to do the `-u` build. The `npm run dist -- -u` isn't really friendly.

File: README.md (hunk 1 above)
Comment (albertjan) [inline README.md]: Navigate to the repository and run `./skulpt.py dist` 
-> 
Navigate to the repository and run `npm run build`

**Comment by j-f1** [comment]
@albertjan Is `npm run build-min` OK?

**Comment by albertjan** [comment]
Yeah sure. I'm gonna wait for input from @skulpt/committers.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -49,28 +49,28 @@ We try to get to pull requests in a very timely way so they don't languish. Noth
...
-* ``Sk.ffi.remapToJs`` and ``Sk.ffi.remapToPy`` are your friends.  They take care of the details 
+
+* ``Sk.ffi.remapToJs`` and ``Sk.ffi.remapToPy`` are your friends.  They take care of the details
 of going from a Python type to a Javascript type and vice versa.  They are smart enough to work
-with common types and even work well recursively converting containers.  ``Sk.ffi.remapToJs`` is 
+with common types and even work well recursively converting containers.  ``Sk.ffi.remapToJs`` is
 **definitely preferred** over  ``foo.v``  
 * Use the ``pyCheckArgs`` function at the beginning of anything that will be exposed to a Python programmer.
 * Check the types of arguments when you know what they must be.
 * Explicitly return ``Sk.builtin.null.null$`` for functions and methds that should return ``None``
```
Comment (waywaaard) [inline CONTRIBUTING.md]: I think, this should be `Sk.builtin.none.none$`. Not related to the PR, but maybe you can just change it anyways.

File: README.md [hunk 3]
```diff
@@ -53,8 +53,8 @@ Building Skulpt is straightforward:
 
 1. Clone the repository from GitHub, ideally using your own fork if you're planning on making any contributions
```
Comment (waywaaard) [inline README.md]: We should add here an hint, that Python 2.7ish is required. The npm commands may delude new users, that no Python is required for building.

Comment (j-f1) [inline README.md]: @waywaaard Done.

File: CONTRIBUTING.md (hunk 2 above)
Comment (j-f1) [inline CONTRIBUTING.md]: @waywaaard Done.

**Comment by albertjan** [comment]
@bnmnetp doesn't really interfere with much but does change the developer interface so I'll wait for an ok from you before merging.

**Comment by bnmnetp** [comment]
Looks to me like this adds to the interface for those that want to use npm, this won't disrupt those who are used to typing ``./skulpt.py dist``. Am I correct?

**Comment by j-f1** [comment]
@bnmnetp You’re correct.

**Pull request merged**